### PR TITLE
[Integration] Remove NodeState singleton

### LINKED ISSUE #114: [Integration] Remove `NodeState` singleton
# Objective
Remove "hacky" code that was introduced during the cross-module integration such as the `NodeState` singleton.

# Origin Document
A lot of tech debt was introduced to maintain the global node state during the cross-module integration stage. This can be see across several files such as `config.go`, `genesis.go` and the `NodeState` struct.

Note that this is a milestone-less issue since it is meant to reduce tech debt on mainline that other branches can rebase on top of.

# Goals / Deliverables
- [ ] Remove the `NodeState` singleton
- [ ] Improve how the node-specific config file is managed & loaded (i.e. see `config.go`)
- [ ] Improve how the node-specific genesis file is managed & loaded (i.e. see `genesis.go`)
- [ ] Consolidate structs / types / interfaces where possible to follow the app-specific bus model

## General issue checklist
- [ ] Update the appropriate CHANGELOG
- [ ] Update the README
- [ ] Update the source code tree explanation
- [ ] Add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)
- [ ] Update any relevant global documentation & references
- [ ] Document small issues / TODOs along the way

# Non-goals
- Add a sort of new behaviour
- Increase configuration validation logic
- Increase genesis validation logic

# Testing Methodology
- Update tests specific to genesis and config files
- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)
Remove

---

Creator: @Olshansk

### PR DESCRIPTION
**Pull request by Olshansk**
# Objective
Remove "hacky" code that was introduced during the cross-module integration such as the `NodeState` singleton.

# Origin Document
Fixes [issue 114](https://github.com/pokt-network/pocket/issues/114): `[Integration] Remove NodeState singleton`

A lot of tech debt was introduced to maintain the global node state during the cross-module integration stage. This can be seen across several files such as `config.go`, `genesis.go` and the `NodeState` struct.

# Main Changes
- Introduced 3 different ways to load the genesis file
- Removed all the code related to `NodeState` and moved it into the consensus module, where appropriate
- Accessing the validator map, block height and app hash through the consensus module
- Removed some parameters that were used as workarounds (e.g. `ValidatorUrlFormat` and `SeedStart`)

# Goals / Deliverables
- [X] Remove the `NodeState` singleton
- [X] Improve how the node-specific config file is managed & loaded (i.e. see `config.go`)
- [X] Improve how the node-specific genesis file is managed & loaded (i.e. see `genesis.go`)
- [ ] Consolidate structs / types / interfaces where possible to follow the app-specific bus model

## General issue checklist
- [ ] Update the appropriate CHANGELOG
- [ ] Update the README
- [ ] Update the source code tree explanation
- [ ] Add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)
- [ ] Update any relevant global documentation & references
- [X] Document small issues / TODOs along the way

# Non-goals
- Add any sort of new behaviour
- Increase configuration/genesis validation logic
- Clean up all of the shared directories

# Testing Methodology
- [x] Added new tests where appropriate
- [X] **All tests**: `make test_all`
- [x] **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)
Remove

### CODE REVIEW

**Comment by Olshansk** [comment]
@andrewnguyen22 @phthan0 PTAL when you have a chance.

**tl;dr Removed the NodeState singleton.**

All the unit tests pass, but there is still a small bug in LocalNet I'm working through. I also have to update the relevant documentation, but otherwise the code is ready for review.

File: consensus/block.go [hunk 1]
```diff
@@ -41,7 +40,7 @@ func (m *consensusModule) prepareBlock() (*types.Block, error) {
 		Height:            int64(m.Height),
 		Hash:              hex.EncodeToString(appHash),
 		NumTxs:            uint32(len(txs)),
-		LastBlockHash:     typesGenesis.GetNodeState(nil).AppHash, // testing temporary
+		LastBlockHash:     m.appHash,
 		ProposerAddress:   m.privateKey.Address(),
 		QuorumCertificate: nil,
 	}
```
Comment (phthan0) [inline consensus/block.go:46]: On line 63 here, do we have a high degree of confidence this call is ok through testing?

Comment (phthan0) [reply consensus/block.go:46]: **_Temporarily accepted._**

**Review by phthan0 — APPROVED** [review]

File: consensus/block.go (hunk 1 above)
Comment (Olshansk) [reply consensus/block.go:46]: TODO(phthan0): Fix it.

**Review by andrewnguyen22 — APPROVED** [review]
